src/sentry/incidents/endpoints/serializers/workflow_engine_detector.py

```diff
@@ -1,34 +1,326 @@
 
-from sentry.api.serializers import Serializer
+from django.contrib.auth.models import AnonymousUser
+
+from sentry.api.serializers import Serializer, serialize
 from sentry.incidents.endpoints.serializers.alert_rule import AlertRuleSerializerResponse
-from sentry.workflow_engine.models import Detector
+from sentry.incidents.endpoints.serializers.workflow_engine_data_condition import (
+    WorkflowEngineDataConditionSerializer,
+)
+from sentry.incidents.endpoints.serializers.workflow_engine_incident import (
+    WorkflowEngineIncidentSerializer,
+)
+from sentry.incidents.models.alert_rule import AlertRuleStatus
+from sentry.models.groupopenperiod import GroupOpenPeriod
+from sentry.sentry_apps.models.sentry_app_installation import prepare_ui_component
+from sentry.sentry_apps.services.app import app_service
+from sentry.sentry_apps.services.app.model import RpcSentryAppComponentContext
+from sentry.snuba.models import SnubaQuery
+from sentry.users.models.user import User
+from sentry.users.services.user.model import RpcUser
+from sentry.users.services.user.service import user_service
+from sentry.workflow_engine.models import (
+    Action,
+    ActionGroupStatus,
+    AlertRuleDetector,
+    DataCondition,
+    DataConditionGroupAction,
+    DataSourceDetector,
+    Detector,
+)
+from sentry.workflow_engine.types import DetectorPriorityLevel
 
 
 class WorkflowEngineDetectorSerializer(Serializer):
-    def serialize(self, obj: Detector, attrs, user, **kwargs) -> AlertRuleSerializerResponse:
-        # TODO: Implement this
-        return {
-            "id": "-2",
-            "name": "Test Alert Rule",
-            "organizationId": "-2",
-            "status": 1,
-            "query": "test",
-            "aggregate": "test",
-            "timeWindow": 1,
-            "resolution": 1,
-            "thresholdPeriod": 1,
-            "triggers": [
-                {
-                    "id": "-1",
-                    "status": 1,
-                    "dateModified": datetime.now(),
-                    "dateCreated": datetime.now(),
-                }
+    """
+    A temporary serializer to be used by the old alert rule endpoints to return data read from the new ACI models
+    """
+
+    def __init__(self, expand: list[str] | None = None, prepare_component_fields: bool = False):
+        self.expand = expand or []
+        self.prepare_component_fields = prepare_component_fields
+
+    def add_sentry_app_installations_by_sentry_app_id(
+        self, actions: list[Action], organization_id: int
+    ) -> Mapping[str, RpcSentryAppComponentContext]:
+        sentry_app_installations_by_sentry_app_id: Mapping[str, RpcSentryAppComponentContext] = {}
+        if self.prepare_component_fields:
+            sentry_app_ids = [
+                int(action.config.get("target_identifier"))
+                for action in actions
+                if action.config.get("sentry_app_identifier") is not None
+            ]
```

Comment:
If the action uses a sentry app then we store the sentry app id as the target identifier - this is different from how it works in the existing alert rule serializer where we pull the sentry app id off of the action directly